src/main/java/macrobase/analysis/outlier/MovingAverage.java

```diff
@@ -0,0 +1,88 @@
+package macrobase.analysis.outlier;
+
+import java.util.Iterator;
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
+
+import org.apache.commons.math3.linear.ArrayRealVector;
+import org.apache.commons.math3.linear.RealVector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import macrobase.datamodel.Datum;
+
+/*
+ * Simple moving average over past `timeSeriesWindowSize` data points.
+ */
+public class MovingAverage extends TimeSeriesOutlierDetector {
+    private static final Logger log = LoggerFactory.getLogger(MovingAverage.class);
+    int prevTime;
+    Queue<Integer> weights;
+    int weightTotal;
+    
+    public MovingAverage(int timeSeriesWindowSize) {
+        super(timeSeriesWindowSize);
+    }
+
+    private RealVector windowSum;
+
+    @Override
+    public void initWindow(Queue<Datum> data) {
+        windowSum = null;
+    }
+
+    @Override
+    public void trainWindow(Queue<Datum> data, Datum newDatum, Datum oldDatum) {
+        /* Empty */
+    }
+
+    @Override
+    public double scoreWindow(Queue<Datum> data, Datum newDatum, Datum oldDatum) {
```

Comment:
For posterity: Peter and I discussed offline and I realized the issue with this `scoreWindow` is essentially that it assumes the window is of a fixed size, and every time you remove a tuple you also add another one. This may not be the case if the window, for example, represents a fixed time range and the points are not uniformly spread out. This can be fixed by having interfaces that allow individually adding/removing one (or many) tuples at a time.